docs/azure/sdk/resource-management.md

```diff
@@ -0,0 +1,326 @@
+// Next we get the specific subscription this resource belongs to
+Subscription subscription = await armClient.GetSubscriptions().GetAsync(id.SubscriptionId);
+// Next we get the specific resource group this resource belongs to
+ResourceGroup resourceGroup = await subscription.GetResourceGroups().GetAsync(id.ResourceGroupName);
+// Finally we get the resource itself
+// Note: for this last stept in this example, Azure.ResourceManager.Compute is needed
+AvailabilitySet availabilitySet = await resourceGroup.GetAvailabilitySets().GetAsync(id.Name);
+```
+
+However, this approach required a lot of code and three API calls to Azure. The same can be done with less code and without any API calls by using extension methods that we have provided on the client itself. These extension methods allow you to pass in a resource identifier and retrieve a scoped client. The object returned is a `[Resource]Operations` mentioned above, since it has not reached out to Azure to retrieve the data yet.
+
+The previous example would end up looking like this:
+
+```csharp
+string resourceId = "/subscriptions/aaaaaaaa-bbbb-cccc-dddd-eeeeeeeeeeee/resourceGroups/workshop2021-rg/providers/Microsoft.Compute/availabilitySets/ws2021availSet";
+// We construct a new armClient to work with
+var armClient = new ArmClient(new DefaultAzureCredential());
+// Next we get the AvailabilitySetOperations object from the client
+// The method takes in a ResourceIdentifier but we can use the implicit cast from string
+AvailabilitySetOperations availabilitySetOperations = armClient.GetAvailabilitySetOperations(resourceId);
+// Now if we want to retrieve the objects data we can simply call get
+AvailabilitySet availabilitySet = await availabilitySetOperations.GetAsync();
+```
+
+### `TryGet` and `DoesExist` convenience methods
+
+If you aren't sure whether a resource you want to get exists, or you just want to check if it exists, use the `TryGet` or `DoesExist` methods. The methods can be invoked from any `[Resource]Container` class.
+
+The `TryGet` and `TryGetAsync` methods return a null object if the specified resource name or ID doesn't exist. On the other hand, the `DoesExist` and `DoesExistAsync` methods return a boolean indicating whether the specified resource exists.
+
+You can find an example for these methods [below](#check-if-resource-group-exists).
+
+## Examples
+
+### Create a resource group
+
+```csharp
+// First, initialize the ArmClient and get the default subscription
+var armClient = new ArmClient(new DefaultAzureCredential());
+// Now we get a ResourceGroup container for that subscription
+Subscription subscription = armClient.DefaultSubscription;
+ResourceGroupContainer rgContainer = subscription.GetResourceGroups();
+
+// With the container, we can create a new resource group with an specific name
+string rgName = "myRgName";
+Location location = Location.WestUS2;
+ResourceGroup resourceGroup = await rgContainer.Construct(location).CreateOrUpdateAsync(rgName);
```

Comment:
we have removed all construct methods for the preview you will need to new up a ResourceGroupData and pass it into Create directly
```suggestion
ResourceGroup resourceGroup = await rgContainer.CreateOrUpdateAsync(rgName, new ResourceGroupData(location));
```